### ISSUE TITLE
Provided Hyperparameters Fail to Reproduce Paper Results

### ISSUE THREAD

**Issue by helloworld751** [jiangxinke/ASD-VAE#3]
![image](https://github.com/user-attachments/assets/5327c157-4a19-479a-a81f-afe5908ff52c)
Hi authors, I followed the configuration as described above, but the reproduction did not achieve the performance reported in your paper.
For instance, using the Cora dataset with structurally missing and a missing rate (mr) of 0.9, the test accuracy I obtained is 0.457, which is significantly lower than the 0.7473 mentioned in the paper. 
Do you have any suggestions? The detailed hyperparameters are attached below.
![image](https://github.com/user-attachments/assets/28cc9137-1711-426c-baee-baaa264a02cb)
![image](https://github.com/user-attachments/assets/c318f88e-29e2-4234-a1b1-c68f7494f25e)
![image](https://github.com/user-attachments/assets/b6332678-9316-417b-8e42-2522b0f0ceae)

**Comment by jiangxinke** [comment]
I argue that some args may not set proper.

Try this args:

Namespace(data='cora', type='struct', rate=0.9, attack_rate=0, nhid=16, dropout=0.4, ncomp=5, lr=0.001, wd=0.01, epoch=4000, patience=20000, verbose='store_true', predict_turns_num=1, discriminator=True, structure_feature='one-hot', cuda=True, hidden=40, enc_name='GCN', r_type='flip', alpha=0.2, loss_alpha_missing=0, FP=20, lambda_gan=1, gan_alpha_missing=10, seed=10242, not_change_adj=True, beta_2=200, use_lcc=False, position_aware_c=10, split_train=True, common_z=True, use_fix=True, profile=False, filter='SGC', dataset='cora')

**Comment by helloworld751** [comment]
Thank you for your quick response. However, the issue persists. 

For clarification, the previous configuration was taken directly from your paper, including settings like 2000 epochs and the Kaze encoder. 

Additionally, I have a minor question: In the configuration you provided, I couldn't find any reference to **'ncomp'** in the latest code commit. Has there been an update that I'm unaware of?

**Config this time:**
Namespace(dataset='cora', type='struct', rate=0.9, verbose='store_true', predict_turns_num=1, cuda=True, lambda_gan=1, use_lcc=False, profile=False, use_fix=True, split_train=True, position_aware_c=10, FP=20, loss_alpha_missing=0, r_type='flip', common_z=True, lr=0.001, wd=0.01, dropout=0.4, epoch=4000, patience=20000, seed=10242, attack_rate=0, structure_feature='one-hot', nhid=16, discriminator=True, hidden=40, enc_name='GCN', alpha=0.2, gan_alpha_missing=10, not_change_adj=True, beta_2=200, filter='SGC')

**Output (last few lines):**
epoch:  3995 train loss: 1.50752 train acc: 0.60714 val loss: 1.76563 val acc: 0.36000
epoch:  3996 train loss: 1.50735 train acc: 0.60714 val loss: 1.76299 val acc: 0.38000
epoch:  3997 train loss: 1.50700 train acc: 0.62143 val loss: 1.76372 val acc: 0.39600
epoch:  3998 train loss: 1.50707 train acc: 0.63571 val loss: 1.76638 val acc: 0.41000
epoch:  3999 train loss: 1.50718 train acc: 0.64286 val loss: 1.76457 val acc: 0.42800
epoch:  4000 train loss: 1.50704 train acc: 0.66429 val loss: 1.75968 val acc: 0.44800
best result: train_loss 1.5137649774551392
train_acc 0.7142857142857143
val_loss 1.747050404548645
val_acc 0.536
test_loss 1.7388402223587036
test_acc 0.544
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [01:02<00:00, 62.46s/it]
0.544
0.0
tpoK: 10, recall: 0.16968287593016576, ndcg: 0.22628675039987348
tpoK: 20, recall: 0.24731993524554152, ndcg: 0.28052592082397126
tpoK: 50, recall: 0.34575509384737707, ndcg: 0.3330659325486798
dataset: cora

**Comment by jiangxinke** [comment]
It seems that your training process has not converaged. (See epoch 4000 & 3999).

I suggest you try more epochs and later I will check my training log and inference details to try my best to help you. But not now, I am busy with the AAAI deadline. (cry)

Good Luck!

**Comment by jiangxinke** [comment]
I've been very busy recently, but now that I have some free time, so I'm addressing this issue again. Sorry for the long wait. 

In the experimental setup, I mentioned setting Katz's beta to 0.667 in the main paper. You can find this code in ‘ASD-VAE/utils/missing.py’ and rewrite the beta value. Please make sure to set the beta in the following code block to 0.667:

```if args.filter == 'Katz':
    data.adj = (data.adj.to_dense() @ data.adj.to_dense() @ data.adj.to_dense() @ data.adj.to_dense()).to_sparse()  \n
 
    beta = 0.667  # rate>=0.7, beta = 0.667
```

I apologize for forgetting to include this in the config. The updated config parameters are:

```
Namespace(dataset='cora', type='struct', rate=0.9, verbose='store_true', predict_turns_num=1, cuda=True, lambda_gan=1, use_lcc=False, profile=False, use_fix=True, split_train=True, position_aware_c=10, FP=20, loss_alpha_missing=0, r_type='flip', common_z=True, lr=0.001, wd=0.01, dropout=0.4, epoch=8000, patience=10000, seed=1024, attack_rate=0.0, structure_feature='one-hot', nhid=16, discriminator=True, hidden=16, enc_name='GAT', alpha=0.2, gan_alpha_missing=10.0, not_change_adj=False, beta_2=200, filter='Katz')
```

Run (1): test ACC is 73.6

![image](https://github.com/user-attachments/assets/fd64094c-bff1-48f4-9820-b5cd5cb8946a)

Run (2): test ACC is 75.8 (which is larger than the results we have reported)
![image](https://github.com/user-attachments/assets/645a0fff-7a50-4f9b-b50d-95873c37add6)


Please feel free to try out my instructions, and I hope my work and code will be helpful to you. Wishing you a pleasant life!

**Comment by cyang810** [comment]
Dear kind author, could you please re-upload the CONFIG file and command so that I can reproduce your results and follow your work!!!

**Comment by jiangxinke** [comment]
> Dear kind author, could you please re-upload the CONFIG file and command so that I can reproduce your results and follow your work!!!

Hello, the CONFIG file is the same with the git version.  You can try and reproduce the results. For any misaligned results, I can help you~
